crates/factor-outbound-mysql/src/client.rs

```diff
@@ -0,0 +1,308 @@
+            Ok(result_set) => {
+                let rows = result_set
+                    .into_iter()
+                    .map(|row| convert_row(row, &columns))
+                    .collect::<Result<Vec<_>, _>>()?;
+
+                Ok(v2_types::RowSet { columns, rows })
+            }
+        }
+    }
+}
+
+fn to_sql_parameter(value: ParameterValue) -> mysql_async::Value {
+    match value {
+        ParameterValue::Boolean(v) => mysql_async::Value::from(v),
+        ParameterValue::Int32(v) => mysql_async::Value::from(v),
+        ParameterValue::Int64(v) => mysql_async::Value::from(v),
+        ParameterValue::Int8(v) => mysql_async::Value::from(v),
+        ParameterValue::Int16(v) => mysql_async::Value::from(v),
+        ParameterValue::Floating32(v) => mysql_async::Value::from(v),
+        ParameterValue::Floating64(v) => mysql_async::Value::from(v),
+        ParameterValue::Uint8(v) => mysql_async::Value::from(v),
+        ParameterValue::Uint16(v) => mysql_async::Value::from(v),
+        ParameterValue::Uint32(v) => mysql_async::Value::from(v),
+        ParameterValue::Uint64(v) => mysql_async::Value::from(v),
+        ParameterValue::Str(v) => mysql_async::Value::from(v),
+        ParameterValue::Binary(v) => mysql_async::Value::from(v),
+        ParameterValue::DbNull => mysql_async::Value::NULL,
+    }
+}
+
+fn convert_columns(columns: Option<Arc<[mysql_async::Column]>>) -> Vec<Column> {
+    match columns {
+        Some(columns) => columns.iter().map(convert_column).collect(),
+        None => vec![],
+    }
+}
+
+fn convert_column(column: &mysql_async::Column) -> Column {
+    let name = column.name_str().into_owned();
+    let data_type = convert_data_type(column);
+
+    Column { name, data_type }
+}
+
+fn convert_data_type(column: &mysql_async::Column) -> DbDataType {
+    let column_type = column.column_type();
+
+    if column_type.is_numeric_type() {
+        convert_numeric_type(column)
+    } else if column_type.is_character_type() {
+        convert_character_type(column)
+    } else {
+        DbDataType::Other
+    }
+}
+
+fn convert_character_type(column: &mysql_async::Column) -> DbDataType {
+    match (column.column_type(), is_binary(column)) {
+        (ColumnType::MYSQL_TYPE_BLOB, false) => DbDataType::Str, // TEXT type
+        (ColumnType::MYSQL_TYPE_BLOB, _) => DbDataType::Binary,
+        (ColumnType::MYSQL_TYPE_LONG_BLOB, _) => DbDataType::Binary,
+        (ColumnType::MYSQL_TYPE_MEDIUM_BLOB, _) => DbDataType::Binary,
+        (ColumnType::MYSQL_TYPE_STRING, true) => DbDataType::Binary, // BINARY type
+        (ColumnType::MYSQL_TYPE_STRING, _) => DbDataType::Str,
+        (ColumnType::MYSQL_TYPE_VAR_STRING, true) => DbDataType::Binary, // VARBINARY type
+        (ColumnType::MYSQL_TYPE_VAR_STRING, _) => DbDataType::Str,
+        (_, _) => DbDataType::Other,
+    }
+}
+
+fn convert_numeric_type(column: &mysql_async::Column) -> DbDataType {
+    match (column.column_type(), is_signed(column)) {
```

Comment:
This is somewhat controversial but I think tiny single-call-site functions should (usually) be inline.

If I need to understand exactly how this code works then I have to scroll down to read the function body, and if I don't need to understand then its just added noise. :shrug: 

```suggestion
    let is_signed = ...;
    match (column.column_type(), is_signed) {
```